src/styles.ts

```diff
@@ -0,0 +1,164 @@
+import * as fs from "fs";
+import {EventEmitter} from "events";
+
+import * as _ from "lodash";
+import * as sass from "node-sass";
+import * as autoprefixer from "autoprefixer";
+import * as cssnano from "cssnano";
+import * as postcss from "postcss";
+import * as styleLint from "stylelint";
+import * as log4js from "log4js";
+
+import config from "./config";
+import utils from "./utils";
+
+export class StyleCompiler extends EventEmitter {
+
+  private logger = log4js.getLogger("global-styles");
+
+  constructor(private entry: string,
+              private output: string,
+              private minify?: boolean,
+              private watch?: string) {
+    super();
+  }
+
+  start() {
+    this.run();
+    if (this.watch) {
+      utils.watch(this.watch, () => this.run());
+    }
+
+    return this;
+  }
+
+  run() {
+    this.compile()
+      .then(
+        () => {
+          this.logger.debug(`Global styles written to ${this.output}`);
+          this.emit("success");
+        },
+        (err) => {
+          this.logger.error("Error processing global styles:", err);
+          this.emit("error");
+        });
+
+    return this;
+  }
+
+  private preprocess(file: string) {
+    return new Promise((resolve, reject) =>
+      sass.render(_.assign({file: file}, config.styles.sass, {
+        sourceMapEmbed: true,
+        sourceMapContents: true,
+      }), (error, result) => {
+        if (error) {
+          error.message = `${error.line}:${error.column} ${error.message}`;
+          reject(error);
+        } else {
+          resolve(result);
+        }
+      }));
+  }
+
+  private postprocess(processed: any) {
+    return new Promise((resolve, reject) => {
+      const processors = [autoprefixer(config.styles.autoprefixer)];
+      if (this.minify) {
+        processors.push(cssnano(config.styles.cssnano));
+      }
+      postcss(processors)
+        .process(processed.css, {map: {inline: false}, to: this.output})
+        .then(
+          (result: any) => resolve(result),
+          reject
+        );
+    });
+  }
+
+  private write(processed: any) {
+    return Promise.all([
+      utils.writeFile(this.output, processed.css),
+      utils.writeFile(`${this.output}.map`, processed.map)
+    ]);
+  }
+
+  private compile() {
+    return this.preprocess(this.entry)
+      .then((processed) => this.postprocess(processed))
+      .then((processed) => this.write(processed));
+  }
+}
+
+export class StyleLinter extends EventEmitter {
+
+  private lintingConfiguration = JSON.parse(fs.readFileSync(".stylelintrc", "utf8"));
```

Comment:
Hm... this path is hard-coded again. It should be possible to configure it, or at least to provide overrides for the rules inside.